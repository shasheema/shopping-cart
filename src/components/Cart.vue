<template>
    <div class="container">
        <h2 class="my-2">Cart <span class="badge badge-secondary">{{cartSize}} products</span></h2>
        <div v-if="!cartSize" class="alert alert-secondary my-5" role="alert">
            Cart is empty! Please add some products.
        </div>
        <table v-else class="table table-striped mb-5">
          <thead class="table-head">
            <tr>
              <th scope="col" id="num-heading">#</th>
              <th scope="col" class="product-column">Product</th>
              <th scope="col">Price</th>
              <th scope="col">Quantity</th>
              <th scope="col">Delete</th>
              <th scope="col">Total</th>
            </tr>
          </thead>
          <tbody>
            <tr class="table-body" v-for="(product, index) in cart" :key="product.id">
              <th class="number" scope="row">{{index + 1}}</th>
              <td>{{product.name}}</td>
              <td>Rs {{product.price.toFixed(2)}}</td>
              <td>
                  <button
                      @click="removeFromCart(product.id)"
                      :disabled="product.quantity === 1"
                      class="btn btn-outline-danger btn-small">
                      -
                  </button>
                  <span class="mx-2">{{product.quantity}}</span>
                  <button
                      @click="addToCart(product.id)"
                      :disabled="product.quantity === product.stock"
                      class="btn btn-outline-success btn-small">
                      +
                  </button>
              </td>
              <td><button
                  @click="deleteFromCart(product.id)"
                  class="btn btn-danger btn-small">
                  Delete
              </button></td>
              <td class="total">Rs {{(product.quantity* product.price).toFixed(2)}}</td>
            </tr>

            <tr class="table-footer">
                <td colspan="5" class="total">Sub Total:</td>
                <td>Rs {{cartTotalAmount.toFixed(2)}}</td>
            </tr>
            <tr class="table-footer">
                <td colspan="5" class="total">Discount:</td>
                <td>Rs
                {{cartTotalAmount>=500 ? cartDiscountAmount.toFixed(2): discount.toFixed(2)}}
                </td>
            </tr>
            <tr class="table-footer">
                <td colspan="5" class="total">VAT:</td>
                <td>Rs {{cartVatAmount.toFixed(2)}}</td>
            </tr>

            <tr class="table-footer">
                <td colspan="5" class="total">Order Total:</td>

                <td>Rs {{ cartTotalAmount>=500 ?
                    ((cartTotalAmount + cartVatAmount) - cartDiscountAmount).toFixed(2):
                    ((cartTotalAmount + cartVatAmount) - discount).toFixed(2)
                    }}</td>
            </tr>
          </tbody>
        </table>
        <router-link to="/checkout" class="btn btn-primary checkout">Checkout</router-link>
    </div>
</template>

<script>
    import {mapState, mapGetters} from 'vuex'
    export default{
        data() {
            return{
                discount: 0,
            };
        },
        computed: {
            ...mapState([
                "cart"
            ]),
            ...mapGetters([
                "cartSize",
                "cartTotalAmount",
                "cartDiscountAmount",
                "cartVatAmount",
            ])
        },
        methods: {
            addToCart(id) {
                this.$store.dispatch("addToCart", id);
            },
            removeFromCart(id) {
                this.$store.dispatch("removeFromCart", id);
            },
            deleteFromCart(id) {
                this.$store.dispatch("deleteFromCart", id);
            }
        }
    }
</script>

<style scoped>
    .table-footer{
        font-size: 1.2em;
        font-weight: bold;
        text-align: right;
    }

    .total{
        text-align: right;
    }
    .table-head{
        background-color: #343a40;
        color: white;
    }
    .checkout{
        float: right;
        color: #FFFFFF;
        text-decoration: none;
        width: 120px;
        height: 30px;
        border-radius: 18px;
        background-color: #047a9c;
        cursor: pointer;
        font-size: 16px;
        box-shadow: none;
        font-weight: 500;
        border: none;
    }
    .btn-danger{
        color: #FFFFFF;
        text-decoration: none;
        width: 70px;
        height: 30px;
        border-radius: 18px;
        cursor: pointer;
        font-size: 12px;
        box-shadow: none;
        font-weight: 500;
        border: none;
    }
</style>
