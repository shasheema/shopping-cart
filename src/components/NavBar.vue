<template>
    <nav class="navbar navbar-expand-sm navbar-dark fixed-top bg-dark">
      <a class="navbar-brand" href="#"><b>My Shop</b></a>

      <div class="collapse navbar-collapse" id="navbarSupportedContent">
        <ul class="navbar-nav mr-auto">
            <li>
             &nbsp;
            </li>
            <li>
                <span class="nav-link"><router-link to="/">Product Catalog
               </router-link></span>
            </li>
            <li>
                <span class="nav-link"><router-link to="/checkout">Shopping Cart
               </router-link></span>
            </li>
        </ul>
      </div>
    </nav>
</template>
