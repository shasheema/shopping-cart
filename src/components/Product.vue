<template>
    <div id="product-page" class="card">
       <div class="card-body">
           <h5 class="card-title">{{ currentProduct.name }}</h5>
           <img :src="currentProduct.image" :alt="currentProduct.name" width="500px">

           <p class="card-text">Price: Rs.<b>{{ currentProduct.price.toFixed(2) }}</b></p>

           <button
                   class="btn mt-3"
                   :class="inCartQuantity ? 'btn-success' :  'btn-danger'"
                   @click="addToCart(currentProduct.id)"
                   :disabled="!inCartQuantity">
               {{inCartQuantity ? "Add to Cart" : "Out of stock"}}
           </button>


       </div>
   </div>
</template>

<script>
    import {mapState, mapGetters, mapActions} from 'vuex'

    export default {
        props: ["product"],
        components: {
        },
        computed: {
            ...mapState([
                "cart"
            ]),
            ...mapGetters({
                currentProduct: "getProduct",
            }),
            inCartQuantity() {
                return this.currentProduct.quantity;
            }
        },
        created(){

            //this.$store.dispatch("fetchProduct");
            let productId = this.$route.params.id;
            this.getProduct(productId);
        },

        methods: {
            ...mapActions([
                'addProduct',
                'showOrHiddenModal',
            ]),

            addToCart(id){
                this.$store.dispatch("addToCart", id);
            },
            getProduct(productId){
                this.$store.dispatch("getProduct", productId);
            }
        },

    };
</script>
<style>
    #product-page.card{
        width: 60%;
        margin: 0 auto;
    }
    .card {
        min-height: 380px;
    }
</style>
