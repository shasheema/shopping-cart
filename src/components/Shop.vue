<template>
    <div id="app">
        <div class="row" id="shop">
            <div class="col-md-7"> <PrductCatalog></PrductCatalog></div>
            <div class="col-md-5"><Cart></Cart></div>
        </div>
    </div>
</template>

<script>
    import PrductCatalog from './ProductCatalog'
    import Cart from './Cart'

    export default {
        name: 'app',
        components: {
            PrductCatalog,
            Cart,
        },

        methods: {
            hideToast() {
                this.$store.commit("hideToast");
            }
        }
    }
</script>

<style>
    #app {
        font-family: 'Avenir', Helvetica, Arial, sans-serif;
        -webkit-font-smoothing: antialiased;
        -moz-osx-font-smoothing: grayscale;
        text-align: left;
        color: #2c3e50;
        margin-top: 60px;
    }
    .row{
        margin-right: 0;
        margin-left: 0;
    }
    #shop .table-footer{
        font-size: 0.7em;
        font-weight: bold;
    }
    #shop .table-body{
        font-size: 0.7em;
    }
    #shop .product-column{
        width: 125px;
    }
    #shop .btn{
        font-size: 12px;
    }
</style>
