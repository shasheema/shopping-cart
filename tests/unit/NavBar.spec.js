import {createLocalVue, mount} from "@vue/test-utils";
import Cart from "@/components/Cart.vue";
import Shop from "@/components/Shop.vue";
import VueRouter from 'vue-router';
import NavBar from "@/components/NavBar.vue";
import route from "../../src/router/index.js";
import Product from "../../src/components/Product";

const localVue = createLocalVue();

localVue.use(VueRouter);

const routes = [{path: '/', component: Shop}, {path: '/checkout', component: Cart}, {path: '/product/:id', component: Product}];
const router = new VueRouter({
    routes
});

const wrapper = mount(route, {
    localVue,
    router
});


expect(wrapper.vm.$route).toBeInstanceOf(Object);

describe("NavBar.vue", () => {

    it('renders without errors', () => {
      expect(NavBar).toMatchSnapshot();
    });
});





