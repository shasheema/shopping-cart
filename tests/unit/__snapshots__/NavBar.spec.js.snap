// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar.vue renders without errors 1`] = `
Object {
  "render": [Function],
  "staticRenderFns": Array [
    [Function],
  ],
}
`;
